akka-actor/src/main/java/akka/annotation/ApiMayChange.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+
+package akka.annotation;
+
+import java.lang.annotation.*;
+
+/**
+ * Marks APIs that are meant to evolve towards becoming stable APIs, but are not stable APIs yet.
+ * 
+ * Evolving interfaces can change from one feature release to another release (i.e. 2.1 to 2.2) without up-front notice.
```

Comment:
We should not change if there is no good reason to change, and changing in a patch release should be done with care, such as deprecating and keeping the old if possible.

However, promising that we will be completely source compatible for all patch releases defeats the purpose of `ApiMayChange`. I would like to leave a door open for doing source incompatible changes also for patch releases, but it should be for exceptional cases.

Normally, I think, it's possible to iterate in patch releases in a source compatible way and then decide that now it's good and then finalize the api and remove the marker and the deprecated cruft. That final decision may also happen within a patch release cycle. The purpose is that we should be able to promote new modules to stable before releasing next 2.x.

Instead of promising things in the documentation related to version numbers it might be better to document this intention/process.